### ISSUE TITLE
Warns about impossible runtime upgrade

### ISSUE THREAD

**Issue by hadess** [flathub-infra/flatpak-builder-lint#1138]
In https://github.com/flathub/io.github.labbots.NiimPrintX/pull/43#issuecomment-5502039896 the linter warned:
> 'runtime-update-available-to-org.freedesktop.Platform-26.08' warning found in linter manifest check. Details: Please consider updating to the latest runtime version

But this package uses the `org.freedesktop.Sdk.Extension.rust-stable` runtime extension which is not available for `26.08`:
```
$ flatpak install --system org.freedesktop.Sdk.Extension.rust-stable/x86_64/26.08
Looking for matches…
error: No remote refs found for ‘org.freedesktop.Sdk.Extension.rust-stable/x86_64/26.08’
```

The linter shouldn't warn about an upgrade that's not actually possible.

**Comment by CodedOre** [comment]
Adding code to check for that is unnecessary complexity.

The maintainers for the SDK extensions will put up branches for the new version when they have time. \
Until then, the warning is not fatal for builds and serves as an reminder to keep an eye on it.

I don't consider this to be actionable.

**Comment by hadess** [comment]
> The maintainers for the SDK extensions will put up branches for the new version when they have time.

Then maybe the warning should only be activated when the majority of SDK extensions are available.

> I don't consider this to be actionable.

It might be quite a bit of work, but it's definitely actionable. I just lost (human) time trying to upgrade to a SDK version that wasn't available...

**Comment by CodedOre** [comment]
The issue is that this would mean more code for something that would only be used for a few days a year. \
More complex code brings more potential for bugs, etc...

I think it would make more sense to consider possibilities for better coordination of the extensions update when a new SDK version is made. 

I'm not saying this isn't an issue, I just think that solving it in the linter is the wrong approach.

**Comment by Erick555** [comment]
Usually extensions are updated when app developers notice they're missing and ask for update. Without update warnings developers won't know about runtime update therefore won't ask for extension update and extensions won't be updated which means no update warnings and so on. There's no 24/7 customer service here.

**Comment by hadess** [comment]
> There's no 24/7 customer service here.

What is this in relation to?

**Comment by Erick555** [comment]
To the _lost time_ complaint. I didn't meant to open discussion here so please ignore that.
